llama_index/vector_stores/chroma.py

```diff
@@ -110,25 +113,31 @@ def from_params(
         port: Optional[str] = None,
         ssl: bool = False,
         headers: Optional[Dict[str, str]] = None,
-        collection_kwargs: Optional[dict] = None,
+        persist_dir: Optional[str] = None,
+        collection_kwargs: Optional[dict] = {},
         **kwargs: Any,
     ) -> "ChromaVectorStore":
         try:
             import chromadb
         except ImportError:
             raise ImportError(import_err_msg)
-
-        client = chromadb.HttpClient(host=host, port=port, ssl=ssl, headers=headers)
-        collection = client.get_or_create_collection(
-            name=collection_name, **collection_kwargs
-        )
-
+        if persist_dir:
+            client = chromadb.PersistentClient(path=persist_dir)
+            collection = client.get_or_create_collection(
+                name=collection_name, **collection_kwargs
+            )
```

Comment:
could we make this an `elif host ...:` (and perhaps also check `port` and `ssl`) then we go thru the `HttpClient` variant.

The `else` would mean that the user hasn't provided specificiations for either options and we should raise an error. For example:

```py
raise ValueError("Please provide either persist_dir or (host, port, ssl).")
```

Reply:
Yeah. That makes sense.